### PR TITLE
Add mock dualtor dataplane test to onboarding t0 job

### PR DESCRIPTION
**Pull request by xwjiang-ms**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
Elastictest performs well in distribute running PR test in multiple KVMs, which support us to add more test scripts to PR checker.
But some traffic test using ptfadapter can't be tested on KVM platform, we need to skip traffic test if needed
#### How did you do it?
Add mock dualtor dataplane test to onboarding t0 job and skip traffic test
#### How did you verify/test it?
Test in KVM platform

### CODE REVIEW

**Comment by xwjiang-ms** [comment]
close PR since previous PR didn't merge

**Comment by xwjiang-ms** [comment]
Reopen since required PR https://github.com/sonic-net/sonic-mgmt/pull/13602 merged

**Review by opcoder0 — APPROVED** [review]
LGTM. Will `test_orch_stress.py` be added to skip separately ?

**Comment by xwjiang-ms** [comment]
> LGTM. Will `test_orch_stress.py` be added to skip separately ?

It's already in PR test t0 job, so needn't to add once more

File: tests/common/dualtor/tunnel_traffic_utils.py [hunk 1]
```diff
@@ -294,6 +295,8 @@ def __enter__(self):
         def __exit__(self, *exc_info):
             if exc_info[0]:
                 return
+            if self.skip_traffic_test is True:
+                return
```
Comment (wangxin) [inline tests/common/dualtor/tunnel_traffic_utils.py:299]: Add some log?

File: tests/common/dualtor/dual_tor_utils.py [hunk 2]
```diff
@@ -1134,6 +1138,8 @@ def check_nexthops_single_downlink(rand_selected_dut, ptfadapter, dst_server_add
     port_packet_count = dict()
     packets_to_send = generate_hashed_packet_to_server(ptfadapter, rand_selected_dut, HASH_KEYS, dst_server_addr,
                                                        expect_packet_num)
+    if skip_traffic_test is True:
+        return
```
Comment (wangxin) [inline tests/common/dualtor/dual_tor_utils.py:1142]: Add some log?

File: tests/common/dualtor/dual_tor_utils.py [hunk 3]
```diff
@@ -869,6 +869,8 @@ def check_tunnel_balance(ptfhost, standby_tor_mac, vlan_mac, active_tor_ip,
     timestamp = datetime.now().strftime('%Y-%m-%d-%H:%M:%S')
     log_file = "/tmp/ip_in_ip_tunnel_test.{}.log".format(timestamp)
     logging.info("PTF log file: %s" % log_file)
+    if skip_traffic_test is True:
+        return
```
[the remark this answers is not in the document]
Comment (wangxin) [reply tests/common/dualtor/dual_tor_utils.py:873]: Another way is to pass it down to `ptf_runner`?

[the remark this answers is not in the document]
Comment (xwjiang-ms) [reply tests/common/dualtor/dual_tor_utils.py]: Fixed, skip early in `check_tunnel_balance`, the reason why still skip inside `check_tunnel_balance` is that we needn't change too much in testcases, but only change 1 or 2 functions

File: tests/common/dualtor/server_traffic_utils.py [hunk 4]
```diff
@@ -120,6 +122,8 @@ def __exit__(self, exc_type, exc_value, traceback):
         logging.info("the expected packet:\n%s", str(self.exp_pkt))
         self.matched_packets = [p for p in self.captured_packets if match_exp_pkt(self.exp_pkt, p)]
         logging.info("received %d matched packets", len(self.matched_packets))
+        if self.skip_traffic_test is True:
+            return
```
[the remark this answers is not in the document]
Comment (xwjiang-ms) [reply tests/common/dualtor/server_traffic_utils.py:127]: Added

File: tests/common/dualtor/tunnel_traffic_utils.py (hunk 1 above)
Comment (xwjiang-ms) [reply tests/common/dualtor/tunnel_traffic_utils.py:300]: Added

File: tests/common/dualtor/dual_tor_utils.py (hunk 2 above)
Comment (xwjiang-ms) [reply tests/common/dualtor/dual_tor_utils.py:1144]: Added

**Review by wangxin — APPROVED** [review]

**Pull request merged**